source/Calamari.Shared/Integration/JsonVariables/JsonConfigurationVariableReplacer.cs

```diff
@@ -182,18 +182,26 @@ private void PopPath()
 
         public void Update(VariableDictionary variables)
         {
-            foreach (var name in variables.GetNames().Where(x => !x.StartsWith("Octopus", StringComparison.OrdinalIgnoreCase)))
+            bool filter(string v)
             {
-                if (map.ContainsKey(name))
+                if (v.StartsWith("Octopus", StringComparison.OrdinalIgnoreCase))
                 {
-                    try
-                    {
-                        map[name](variables.Get(name));
-                    }
-                    catch (Exception e)
-                    {
-                        Log.WarnFormat("Unable to set value for {0}. The following error occurred: {1}", name, e.Message);
-                    }
+                    // Only include variables starting with 'Octopus'
+                    // if it also has a colon (:)
+                    return v.StartsWith("Octopus:");
+                }
+                return map.ContainsKey(v);
```

Comment:
I think there is a logic gap here in the filter method. In the case where it starts with `Octopus:` should we not then be doing a `map.ContainsKey(v)`? I was expecting to just see something like ` || x.StartsWith("Octopus:", StringComparisn.OrdinalIgnoredCase` added to the original Where clause